cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/SparkPipelineRunner.java

```diff
@@ -551,6 +552,28 @@ private Optional<EmittedRecords> tryRelationalTransform(
     }
 
     if (plugin instanceof RelationalTransform) {
+      // If there are multiple inputs to a "LinearRelationalTransform" implemented stage, then combine all the inputs
+      // into 1 dataset, as LinearRelationalTransform expects only 1 input.
+      if ((plugin instanceof LinearRelationalTransform) && inputDataCollections.size() > 1) {
+        Iterator<String> keys = inputDataCollections.keySet().iterator();
+        String firstKey = keys.next();
+        Schema firstSchema = stageSpec.getInputSchemas().get(firstKey);
+        SparkCollection<Object> first = inputDataCollections.get(firstKey);
+
+        while (keys.hasNext()) {
+          String currKey = keys.next();
+          //Validate schema is Same for all
+          if (firstSchema.equals(stageSpec.getInputSchemas().get(currKey))) {
+            first = first.union(inputDataCollections.get(currKey));
+          } else {
+            //If there is schema mismatch, return empty option (handle via usual cdap flow)
+            Optional.empty();
+          }
+        }
+        inputDataCollections.clear();
+        inputDataCollections.put(firstKey, first);
```

Comment:
My only nit here is that the input key name should be something like "<the_stage_name>_combinedinput" so it's obvious what this stage represents.

Reply:
Actually I had thought on similar lines , but in the following stages it tried to Pull in the Schema based on this KEY name. [code](https://github.com/cdapio/cdap/blob/b54e1fb164d234ea25798e35a214cc214eb69967/cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java#L799)

I didn't want to modify the `StageSpec` object for this. 

Let me know your thoughts.

Reply:
Ah, this makes a lot of sense now. Thanks for clarifying!